core/src/impl/Kokkos_Error.hpp

```diff
@@ -17,99 +17,20 @@
 #ifndef KOKKOS_IMPL_ERROR_HPP
 #define KOKKOS_IMPL_ERROR_HPP
 
-#include <new>  // align_val_t
 #include <string>
-#include <iosfwd>
 #include <Kokkos_Macros.hpp>
```

Comment:
You only need to include `<string>, the other includes are not necessary in the header file.

Reply:
You're correct, but a lot of headers relied on these transitive includes and I did not really want to fix it here.
Is that ok to leave it as future cleanup work?

Reply:
Yes, we can do it another time.